nimbus/src/enrollment.rs

```diff
@@ -20,28 +24,392 @@ pub enum EnrolledReason {
     OptIn,     // Explicit opt-in.
 }
 
+// These are types we use internally for managing non-enrollments.
+#[derive(Deserialize, Serialize, Debug, Clone, Hash, Eq, PartialEq)]
+pub enum NotEnrolledReason {
+    OptOut,      // The user opted-out of experiments before we ever got enrolled to this one.
+    NotSelected, // The evaluator bucketing did not choose us.
+    NotTargeted, // We are not being targeted for this experiment.
+    EnrollmentsClosed, // The experiment enrollment is paused.
```

Comment:
Minor nit: calling this `EnrollmentsPaused` would line up better with the docstring, the name of the field in remote settings, etc.